include/umf/providers/provider_level_zero.h

```diff
@@ -14,6 +14,9 @@
 extern "C" {
 #endif
 
+typedef struct _ze_context_handle_t *ze_context_handle_t;
```

Comment:
I'm not we should create typedefs defined in ze_api.h - there would be no conflict if user include this file?

Reply:
There will be no problem with include, this is already what's happening in provider_level_zero.c. But I wonder if this typdef can change in the future - would that be considered L0 API breaking change or would this be allowed?

Reply:
@vinser52 any thoughts? should we just keep void* here?

Reply:
hmm, the only concern I have is what if this header and L0 header are used simultaneously in the user code.

As @igchor mentioned above there is already such situation in `provider_level_zero.c` source file. It would be good also to check with C++ compiler. Our code is plain C.

Reply:
This is a regular forward declaration, there's nothing here that could cause problems. I checked with the C++ compiler as well and everything works fine.